### ISSUE TITLE
timescaledb 0.12 crashed frequently on Ubuntu 18.04

### ISSUE THREAD

**Issue by abcfy2** [timescale/timescaledb#727]
Here is my postgresql log, you can find the timecaledb crashed frequently(Seems every 3 minutes.):

```
2018-09-18 15:20:36.967 CST [8347] 提示:  You may start background workers manually by using the  _timescaledb_internal.start_background_workers() function in each database you would like to have a scheduler worker in.
2018-09-18 15:20:36.967 CST [8347] 错误:  TimescaleDB background worker internal error: cannot decrement workers below 1
2018-09-18 15:20:36.967 CST [8347] 提示:  The background worker scheduler is in an invalid state, please submit a bug report.
2018-09-18 15:20:36.974 CST [891] 日志:  工作进程: TimescaleDB Background Worker Launcher (PID 8347) 已退出, 退出代码 1
2018-09-18 15:23:53.279 CST [8493] 比致命错误还过分的错误:  stuck spinlock detected at bgw_counter_reinit, /build/timescaledb-postgresql-10-aGDrQt/timescaledb-postgresql-10-0.12.0~ubuntu18.04/src/loader/bgw_counter.c:95
2018-09-18 15:23:53.574 CST [891] 日志:  工作进程: TimescaleDB Background Worker Launcher (PID 8493) 被信号 6 中断: Aborted
2018-09-18 15:23:53.574 CST [891] 日志:  中断任何其它已激活的服务器进程
2018-09-18 15:23:53.580 CST [8345] 警告:  中断联接, 因为其它服务器进程崩溃
2018-09-18 15:23:53.580 CST [8345] 详细信息:  Postmaster 命令此服务器进程回滚当前事物并退出, 因为其它服务器进程不正常的退出可能毁坏了共享内存.
2018-09-18 15:23:53.580 CST [8345] 提示:  一会儿你将可以重联接数据库并且重复你的命令.
2018-09-18 15:23:53.582 CST [891] 日志:  所有的服务器进程被中止; 重新初始化
2018-09-18 15:23:53.608 CST [8906] 日志:  数据库系统中断；上一次的启动时间是在2018-09-18 15:20:36 CST
2018-09-18 15:23:53.775 CST [8906] 日志:  数据库系统没有正确的关闭; 处于自动恢复状态中
2018-09-18 15:23:53.785 CST [8906] 日志:  redo 在 0/67319630 开始
2018-09-18 15:23:53.785 CST [8906] 日志:  0/67319668 处的记录长度不合法：想要 24，但得到的是 0
2018-09-18 15:23:53.785 CST [8906] 日志:  redo 在 0/67319630 完成
2018-09-18 15:23:53.869 CST [891] 日志:  数据库系统准备接受连接
2018-09-18 15:23:53.869 CST [8919] 日志:  TimescaleDB background worker launcher connected to shared catalogs
2018-09-18 15:23:53.869 CST [8919] 日志:  total databases = 13 TimescaleDB background worker limit 8
2018-09-18 15:23:53.869 CST [8919] 提示:  You may start background workers manually by using the  _timescaledb_internal.start_background_workers() function in each database you would like to have a scheduler worker in.
2018-09-18 15:23:53.870 CST [8919] 错误:  TimescaleDB background worker internal error: cannot decrement workers below 1
2018-09-18 15:23:53.870 CST [8919] 提示:  The background worker scheduler is in an invalid state, please submit a bug report.
2018-09-18 15:23:53.876 CST [891] 日志:  工作进程: TimescaleDB Background Worker Launcher (PID 8919) 已退出, 退出代码 1
2018-09-18 15:27:09.363 CST [9136] 比致命错误还过分的错误:  stuck spinlock detected at bgw_counter_reinit, /build/timescaledb-postgresql-10-aGDrQt/timescaledb-postgresql-10-0.12.0~ubuntu18.04/src/loader/bgw_counter.c:95
2018-09-18 15:27:09.645 CST [891] 日志:  工作进程: TimescaleDB Background Worker Launcher (PID 9136) 被信号 6 中断: Aborted
2018-09-18 15:27:09.645 CST [891] 日志:  中断任何其它已激活的服务器进程
2018-09-18 15:27:09.654 CST [8917] 警告:  中断联接, 因为其它服务器进程崩溃
2018-09-18 15:27:09.654 CST [8917] 详细信息:  Postmaster 命令此服务器进程回滚当前事物并退出, 因为其它服务器进程不正常的退出可能毁坏了共享内存.
2018-09-18 15:27:09.654 CST [8917] 提示:  一会儿你将可以重联接数据库并且重复你的命令.
2018-09-18 15:27:09.658 CST [891] 日志:  所有的服务器进程被中止; 重新初始化
2018-09-18 15:27:09.685 CST [9429] 日志:  数据库系统中断；上一次的启动时间是在2018-09-18 15:23:53 CST
2018-09-18 15:27:09.829 CST [9429] 日志:  数据库系统没有正确的关闭; 处于自动恢复状态中
2018-09-18 15:27:09.858 CST [9429] 日志:  redo 在 0/673196D8 开始
2018-09-18 15:27:09.858 CST [9429] 日志:  0/67319710 处的记录长度不合法：想要 24，但得到的是 0
2018-09-18 15:27:09.858 CST [9429] 日志:  redo 在 0/673196D8 完成
2018-09-18 15:27:09.946 CST [891] 日志:  数据库系统准备接受连接
2018-09-18 15:27:09.946 CST [9440] 日志:  TimescaleDB background worker launcher connected to shared catalogs
2018-09-18 15:27:09.947 CST [9440] 日志:  total databases = 13 TimescaleDB background worker limit 8
2018-09-18 15:27:09.947 CST [9440] 提示:  You may start background workers manually by using the  _timescaledb_internal.start_background_workers() function in each database you would like to have a scheduler worker in.
2018-09-18 15:27:09.947 CST [9440] 错误:  TimescaleDB background worker internal error: cannot decrement workers below 1
2018-09-18 15:27:09.947 CST [9440] 提示:  The background worker scheduler is in an invalid state, please submit a bug report.
2018-09-18 15:27:09.954 CST [891] 日志:  工作进程: TimescaleDB Background Worker Launcher (PID 9440) 已退出, 退出代码 1
2018-09-18 15:30:25.325 CST [9659] 比致命错误还过分的错误:  stuck spinlock detected at bgw_counter_reinit, /build/timescaledb-postgresql-10-aGDrQt/timescaledb-postgresql-10-0.12.0~ubuntu18.04/src/loader/bgw_counter.c:95
2018-09-18 15:30:25.595 CST [891] 日志:  工作进程: TimescaleDB Background Worker Launcher (PID 9659) 被信号 6 中断: Aborted
2018-09-18 15:30:25.595 CST [891] 日志:  中断任何其它已激活的服务器进程
2018-09-18 15:30:25.603 CST [9438] 警告:  中断联接, 因为其它服务器进程崩溃
2018-09-18 15:30:25.603 CST [9438] 详细信息:  Postmaster 命令此服务器进程回滚当前事物并退出, 因为其它服务器进程不正常的退出可能毁坏了共享内存.
2018-09-18 15:30:25.603 CST [9438] 提示:  一会儿你将可以重联接数据库并且重复你的命令.
2018-09-18 15:30:25.605 CST [891] 日志:  所有的服务器进程被中止; 重新初始化
2018-09-18 15:30:25.629 CST [9997] 日志:  数据库系统中断；上一次的启动时间是在2018-09-18 15:27:09 CST
2018-09-18 15:30:25.752 CST [9997] 日志:  数据库系统没有正确的关闭; 处于自动恢复状态中
2018-09-18 15:30:25.765 CST [9997] 日志:  redo 在 0/67319780 开始
2018-09-18 15:30:25.765 CST [9997] 日志:  0/673197B8 处的记录长度不合法：想要 24，但得到的是 0
2018-09-18 15:30:25.765 CST [9997] 日志:  redo 在 0/67319780 完成
2018-09-18 15:30:25.837 CST [891] 日志:  数据库系统准备接受连接
2018-09-18 15:30:25.837 CST [10009] 日志:  TimescaleDB background worker launcher connected to shared catalogs
2018-09-18 15:30:25.837 CST [10009] 日志:  total databases = 13 TimescaleDB background worker limit 8
2018-09-18 15:30:25.837 CST [10009] 提示:  You may start background workers manually by using the  _timescaledb_internal.start_background_workers() function in each database you would like to have a scheduler worker in.
2018-09-18 15:30:25.838 CST [10009] 错误:  TimescaleDB background worker internal error: cannot decrement workers below 1
2018-09-18 15:30:25.838 CST [10009] 提示:  The background worker scheduler is in an invalid state, please submit a bug report.
2018-09-18 15:30:25.845 CST [891] 日志:  工作进程: TimescaleDB Background Worker Launcher (PID 10009) 已退出, 退出代码 1
```

I'm using the Ubuntu 18.04 with the latest version of postgresql (10.5):

```
 postgres=# SELECT version();
                                                          version                                                           
----------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 10.5 (Ubuntu 10.5-1.pgdg18.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 7.3.0-16ubuntu3) 7.3.0, 64-bit
(1 行记录)
```

I installed postgresql from official repo: http://apt.postgresql.org, and I install the timescaledb from ppa: https://launchpad.net/~timescale/+archive/ubuntu/timescaledb-ppa

What's wrong with me ? Any help?

**Comment by amytai** [comment]
Thanks for reporting this! We are aware of the error and are working on a fix that will be out later this week. In the meantime, can you set your configuration settings to account for your 13 databases? In particular, `timescaledb.max_background_workers` must be at least 14, and your other Postgres configs should be able to accommodate 14. You can read more here: https://docs.timescale.com/v1.0/getting-started/configuring#workers

Also, are you seeing an error when you run your SQL commands? These errors in the log are coming from a part of the TimescaleDB extension that should not affect your actual database or data.

**Comment by abcfy2** [comment]
Thanks. If this is a very impotant config, I hope this should be mentioned in the [installation](https://docs.timescale.com/v1.0/getting-started/installation/linux/installation-source) or [setup](https://docs.timescale.com/v1.0/getting-started/setup) document.

**Comment by amytai** [comment]
Thanks for the feedback. We're working on the best way to have users interact with these configs.. stay tuned.

**Comment by mfreed** [comment]
This should be fixed in the 0.12.1 release:  https://github.com/timescale/timescaledb/releases/tag/0.12.1

Please let us know if you continue to have problems.

**Comment by abcfy2** [comment]
Thanks. It seems this issue has been solved.
